build/azure-pipelines/sql-product-compile.yml

```diff
@@ -122,15 +112,17 @@ steps:
     node build/azure-pipelines/common/copyArtifacts.js
   displayName: Write Version Information
 
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish Artifact: drop'
+- script: |
+    set -e
+    tar -czf $(Build.ArtifactStagingDirectory)/compilation.tar.gz .build out-*
+  condition: and(succeeded(), ne(variables.COMPILATION_RESTORED, 'true'))
```

Comment:
COMPILATION_RESTORED doesn't exist anymore so you should be able to just get rid of this